src/receive_imf.rs

```diff
@@ -703,7 +703,9 @@ async fn add_parts(
         better_msg = Some(stock_str::msg_location_enabled_by(context, from_id).await);
     }
 
-    let parent = get_parent_message(context, mime_parser).await?;
+    let parent = get_parent_message(context, mime_parser)
+        .await?
+        .filter(|p| Some(p.id) != replace_msg_id);
```

Comment:
Note that this may lead to the message suddenly disappear after clicking the "Download" button, which may be equally confusing. OTOH, we can just merge it and see whether people complain, and then think about which behavior is better (or whether whether we should even send a toast "Oops, this message actually belongs to group XY, fixed it now", which ofc would need support from the UIs)